components/tidb_query_datatype/src/codec/data_type/vector.rs

```diff
@@ -37,6 +37,47 @@ impl VectorValue {
         }
     }
 
+    /// Creates a `VectorValue` of length `len` with the given value `scalar`.
+    #[inline]
+    pub fn from_scalar(scalar: &ScalarValue, len: usize) -> Self {
+        macro_rules! expand_convertion {
+            ($val:tt, $( $tp:tt : $chktp:ty ),* ) => {
+                match &$val {
+                    $(
+                        &ScalarValue::$tp(val) => {
+                            let mut v: $chktp = ChunkedVec::with_capacity(len);
```

Comment:
Is this pattern common? I found it's the same with `v.push(val)` logicly.

The difference is code here only need to `match` val once, but `v.push(val)` will check it everytime.

If this pattern is common, maybe we can implement a `pushs` or `push_multiple`?

ping @skyzh for advice.

Reply:
I believe this is a rare case. We could leave this as-is.

Reply:
probably we can refactor it when needed, but for now I think it is okay.